system/jlib/jlzw.cpp

```diff
@@ -2102,14 +2110,16 @@ class CCompressedFile : public CInterface, implements ICompressedFileIO
                 src += len;
             }
         }
-        else 
+        else // lzw or fastlz or lz4
+        {
             src += compressor->write(src, len);
+        }
         return (size32_t)(src-(const byte *)expbuf);
     }
 public:
     IMPLEMENT_IINTERFACE;
 
-    CCompressedFile(IFileIO *_fileio,IMemoryMappedFile *_mmfile,CompressedFileTrailer &_trailer,ICFmode _mode, bool _setcrc,ICompressor *_compressor,IExpander *_expander, bool fast)
+    CCompressedFile(IFileIO *_fileio,IMemoryMappedFile *_mmfile,CompressedFileTrailer &_trailer,ICFmode _mode, bool _setcrc,ICompressor *_compressor,IExpander *_expander, __int64 _compMethod)
```

Comment:
As part of this re-factoring, can we not standardize methods/classes that compress (like this class) on requiring a ICompressor (or IExpander) ?
I think a lot of the conditional special casing code could disappear if so, you'd probably need a couple more methods in the interfaces to supply type or other compressor specific information, but not much.

I see that CCompressedFile implements it's own row diff, any idea why it's not using createRDiffCompressor()?